src/Server/modules/WorldPlaceManager.js

```diff
@@ -0,0 +1,245 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+
+// Array of WorldPlaces read from CSV
+var allWorldPlace = [];
+
+// Populates allWorldPlace with data read from CSV
+function InitializeAllWorldPlace() {
+    allWorldPlace = readCSVModule.LoadWorldData();
+}
+
+function AddallWorldPlace(province, country, date, tod) {
+    if (allWorldPlace == undefined) InitializeAllWorldPlace();
+
+    // Copied from ModifyData.js
+    var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allWorldPlace.length; ++i) {
+			if (country == allWorldPlace[i].country && province == allWorldPlace[i].state) {
+				for (var key in allWorldPlace[i].numConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allWorldPlace[i].numConfirmed[temp_date] = number;
+				break;
+			} else if (i == allWorldPlace.length - 1) {
+				errormsg = 'wrong place';
+				break;
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allWorldPlace.length; ++i) {
+			if (country == allWorldPlace[i].country && province == allWorldPlace[i].state) {
+				for (var key in allWorldPlace[i].numDeaths) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allWorldPlace[i].numDeaths[temp_date] = number;
+				break;
+			} else if (i == allWorldPlace.length - 1) {
+				errormsg = 'wrong place';
+				break;
+			}
+		}
+	} else if (tod == 3) {
+		for (var i = 0; i < allWorldPlace.length; ++i) {
+			if (country == allWorldPlace[i].country && province == allWorldPlace[i].state) {
+				for (var key in allWorldPlace[i].numRecovered) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allWorldPlace[i].numRecovered[temp_date] = number;
+				break;
+			} else if (i == allWorldPlace.length - 1) {
+				errormsg = 'wrong place';
+				break;
+			}
+		}
+	}
+	return errormsg;
+}
+
+function EditallWorldPlace(province, country, date, tod, number) {
```

Comment:
When we edit data, we also need to update currentNumConfirmed and two other variables.